packages/expo-clipboard/ios/ApplePasteButton.swift

```diff
@@ -0,0 +1,169 @@
+      control.bottomAnchor.constraint(equalTo: bottomAnchor),
+      control.leadingAnchor.constraint(equalTo: leadingAnchor),
+      control.trailingAnchor.constraint(equalTo: trailingAnchor)
+    ])
+  }
+
+  override func paste(itemProviders: [NSItemProvider]) {
+    if #available(iOS 14.0, *) {
+      for provider in itemProviders {
+        if provider.hasItemConformingToTypeIdentifier(UTType.image.identifier) {
+          _ = provider.loadObject(ofClass: UIImage.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted image")
+              return
+            }
+            self.processImage(item: data)
+          }
+        } else if provider.hasItemConformingToTypeIdentifier(UTType.url.identifier) {
+          _ = provider.loadObject(ofClass: URL.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted content")
+              return
+            }
+
+            self.onPastePressed([
+              "type": "text",
+              "text": data?.absoluteString
+            ])
+          }
+        } else if provider.hasItemConformingToTypeIdentifier(UTType.html.identifier) && acceptedContentTypes.contains(.html) {
+          _ = provider.loadObject(ofClass: String.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted content")
+              return
+            }
+            self.processHtml(data: data)
+          }
+        } else if provider.hasItemConformingToTypeIdentifier(UTType.utf8PlainText.identifier) && !acceptedContentTypes.contains(.html) {
+          _ = provider.loadObject(ofClass: String.self) { data, error in
+            guard error == nil else {
+              log.error("Error loading pasted content")
+              return
+            }
+
+            guard let data = data as? String else {
+              log.error("Failed to read text data")
+              return
+            }
+
+            self.onPastePressed([
+              "type": "text",
+              "text": data
+            ])
+          }
+        }
+      }
+    }
+  }
+
+  @available(iOS 14.0, *)
+  private func setContentTypes() {
+    if acceptedContentTypes.isEmpty {
+      pasteConfiguration = UIPasteConfiguration(acceptableTypeIdentifiers: [
+        UTType.utf8PlainText.identifier,
+        UTType.image.identifier
+      ])
+    } else {
+      pasteConfiguration = UIPasteConfiguration(acceptableTypeIdentifiers: acceptedContentTypes.map {
+        $0.typeIdentifier()
+      })
+    }
+  }
+
+  private func processImage(item: NSItemProviderReading?) {
+    guard let image = item as? UIImage else {
+      log.error("Failed to read image data")
+      return
+    }
+
+    guard let data = imageToData(image) else {
+      log.error("Failed to process image data")
+      return
+    }
+
+    let imageData = "data:\(imageOptions.imageFormat.getMimeType());base64,\(data.base64EncodedString())"
```

Comment:
Hmm, sending it as base64 might not be a good choice, especially for large images. I would rather save it to the file system, but ideally we will just return a shared reference to this image.

Reply:
I just kept the same implementation that we have when using the Pasteboard API. Could save it to the cache directory first.

Reply:
I see. That's not good, but ok, let's keep it consistent then.

Reply:
I've changed it but I can do it in both places?